keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -638,24 +640,201 @@ always dispatched immediately.  Following is how the other requests
+  // levels borrowing seats from this level.  This does not limit dispatching from
+  // this priority level that borrows seats from lower priority levels (those lower
+  // levels do that).  The server's concurrency limit (ServerCL) is divided among the
+  // Limited priority levels in proportion to their ACS values:
+  //
+  // NominalCL(i)  = ceil( ServerCL * ACS(i) / sum_acs )
+  // sum_acs = sum[limited priority level k] ACS(k)
+  //
+  // Bigger numbers mean a larger nominal concurrency limit, at the expense
+  // of every other Limited priority level.
+  // This field has a default value of 30.
+  // +optional
+  AssuredConcurrencyShares int32
+
+  // `borrowablePercent` prescribes the fraction of the level's NominalCL that
+  // can be borrowed by higher priority levels.  This value of this
+  // field must be between 0 and 100, inclusive, and it defaults to 0.
+  // The number of seats that higher levels can borrow from this level, known
+  // as this level's BorrowableConcurrencyLimit (BorrowableCL), is defined as follows.
+  //
+  // BorrowableCL(i) = round( NominalCL(i) * borrowablePercent(i)/100.0 )
+  //
+  // +optional
+  BorrowablePercent int32
+
+  // `priority` determines where this priority level appears in the total order
+  // of Limited priority levels used to configure borrowing between those levels.
+  // A numerically higher value means a logically lower priority.
+  // Do not create ties; they will be broken arbitrarily.
+  // `priority` MUST BE between 0 and 10000, inclusive, and
+  // SHOULD BE greater than zero.
+  // If it is zero then, for the sake of a smooth transition from the time
+  // before this field existed, this level will be treated as if its `priority`
+  // is the average of the `matchingPrecedence` of the FlowSchema objects
+  // that reference this level.
+  // +optional
+  Priority int32
+}
 ```
 
-where SCL is the apiserver's concurrency limit and ACS(l) is the
-AssuredConcurrencyShares for priority level l.
+Prior to the introduction of borrowing, the `assuredConcurrencyShares`
+field had two meanings that amounted to the same thing: the total
+shares of the level, and the non-borrowable shares of the level.
+While it is somewhat unnatural to keep the meaning of "total shares"
+for a field named "assured" shares, rolling out the new behavior into
+existing systems will be more continuous if we keep the meaning of
+"total shares" for the existing field.  In the next version we should
+rename the `AssuredConcurrencyShares` to `NominalConcurrencyShares`.
+
+Consider rolling the borrowing behavior into a pre-existing cluster
+that did not have borrowing.  In particular, suppose that the
+administrators of the cluster have scripts/clients that maintain some
+out-of-tree PriorityLevelConfiguration objects; these, naturally, do
+not specify a value for the `priority` field.  The default behavior
+for `priority` is designed to do something more natural and convenient
```

Comment:
This comment isn't up-to-date I think, as we don't use `priority` field at all.

Remove?

Reply:
Yes, I deleted it as part of removing the Priority field.